### PR TITLE
Parallelize puppet and chef tests

### LINKED ISSUE #72: Run rspec tests in parallel
Investigate if parralel_tests can be used to speed up execution of rspec tests.

Ensure the following systems are using parallel_tests:

- [x] Magic modules
- [ ] Puppet
- [ ] Chef

Installation: https://rubygems.org/gems/parallel_tests

On a cursory tests it improved puppet/compute unit tests execution from 435s (07:15) to 104s (01:04).

### PR DESCRIPTION
**Pull request by rosbo**
Fixes #72

The chefs tests were not being run because of the `SPEC_DIR` issue. See #236 for more details.
I fixed the `SPEC_DIR` issue but a lot of chef-tests are failing. I created a separate issue to fix and enable the chef-tests (#236).

Note: I also fixed some chef tests.

-----------------------------------------------------------------
# [all]

### CODE REVIEW

**Comment by ndmckinley** [comment]
Shouldn't need to do anything but this - if this made it through Concourse, that would be how you'd know it's correct.  I see it didn't, so let's see why.

**Comment by rosbo** [comment]
`puppet-test` are now running in ~3 minutes. 3x improvement!

File: templates/chef/tests/absent~no_action.erb [hunk 1]
```diff
@@ -44,10 +44,9 @@ let(:runner) do
   # add in the mocked version so that the tests do not fail.
   # Since cookbooks can have any name, we assume that
   # any directory with the word auth is the google-gauth cookbook.
-  if Dir.entries(parent_dir).select { |p| p.include? 'auth' }.empty?
-    cookbook_paths << File.join(File.dirname(__FILE__), 'cookbooks')
-  end
-  ChefSpec::SoloRunner.new(
+  cookbook_paths << File.join(File.dirname(__FILE__), 'cookbooks')
+
+ChefSpec::SoloRunner.new(
```
Comment (rambleraptor) [inline templates/chef/tests/absent~no_action.erb]: Indentation issue.

File: templates/chef/tests/absent~delete.erb [hunk 2]
```diff
@@ -46,10 +46,9 @@ let(:runner) do
   # add in the mocked version so that the tests do not fail.
   # Since cookbooks can have any name, we assume that
   # any directory with the word auth is the google-gauth cookbook.
-  if Dir.entries(parent_dir).select { |p| p.include? 'auth' }.empty?
-    cookbook_paths << File.join(File.dirname(__FILE__), 'cookbooks')
-  end
-  ChefSpec::SoloRunner.new(
+  cookbook_paths << File.join(File.dirname(__FILE__), 'cookbooks')
```
Comment (rambleraptor) [inline templates/chef/tests/absent~delete.erb]: Indentation issue

**Pull request merged**